src/Xamarin.Android.Build.Tasks/Tasks/LinkApplicationSharedLibraries.cs

```diff
@@ -0,0 +1,260 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using ThreadingTasks = System.Threading.Tasks;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using Xamarin.Android.Tools;
+
+namespace Xamarin.Android.Tasks
+{
+	public class LinkApplicationSharedLibraries : AsyncTask
+	{
+		sealed class Config
+		{
+			public string LinkerPath;
+			public string LinkerOptions;
+			public string OutputSharedLibrary;
+		}
+
+		sealed class InputFiles
+		{
+			public List<string> ObjectFiles;
+			public string OutputSharedLibrary;
+		}
+
+		[Required]
+		public ITaskItem[] ObjectFiles { get; set; }
+
+		[Required]
+		public ITaskItem[] ApplicationSharedLibraries { get; set; }
+
+		[Required]
+		public bool DebugBuild { get; set; }
+
+		[Required]
+		public string AndroidNdkDirectory { get; set; }
+
+		public override bool Execute ()
+		{
+			try {
+				return DoExecute ();
+			} catch (Exception e) {
+				Log.LogCodedError ("XA3001", $"{e}");
```

Comment:
Same here: it's user-hostile to dump a stack trace into the Errors window (what is the user supposed to do with a stack trace?), and the message is probably not context free.

```suggestion
				Log.LogCodedError ("XA3001", $"Unable to link `libxamarin-app.so`: {e.Message}");
				Log.LogDebugMessage (e.ToString ());
```